builtin/logical/pki/storage.go

```diff
@@ -552,7 +556,27 @@ func (sc *storageContext) fetchIssuerById(issuerId issuerID) (*issuerEntry, erro
 		return nil, errutil.InternalError{Err: fmt.Sprintf("unable to decode pki issuer with id %s: %v", issuerId.String(), err)}
 	}
 
-	return &issuer, nil
+	return sc.upgradeIssuerIfRequired(&issuer), nil
+}
+
+func (sc *storageContext) upgradeIssuerIfRequired(issuer *issuerEntry) *issuerEntry {
+	// *NOTE*: Don't attempt to write out the issuer here as it may cause ErrReadOnly that will direct the
```

Comment:
Kit says we have the backend stuff that we could use to infer if we're able to do the write and avoid the forwarding.

Reply:
While implementing this, it dawned on me that we don't know if we have the rw issuers lock so this wouldn't be safe to do at this level.